app/models/ems_event.rb

```diff
@@ -58,14 +58,46 @@ def self.group_name(group)
     group[:name]
   end
 
-  def self.add_queue(meth, ems_id, event)
-    MiqQueue.submit_job(
-      :service     => "event",
-      :target_id   => ems_id,
-      :class_name  => "EmsEvent",
-      :method_name => meth,
-      :args        => [event],
-    )
+  def self.artemis_client
+    require "manageiq-messaging"
+    # FIXME: logger should be set at a global level
+    ManageIQ::Messaging.logger = _log
+    @artemis_client ||= begin
+      queue_settings = Settings.workers.worker_base.queue_worker_base.event_handler
+      connect_opts = {
+        :host       => queue_settings.queue_hostname,
+        :port       => queue_settings.queue_port.to_i,
+        :username   => queue_settings.queue_username,
+        :password   => queue_settings.queue_password,
+        :client_ref => "event_handler",
+      }
+
+      # caching the client works, even if the connection becomes unavailable
+      # internally the client will track the state of the connection and re-open it,
+      # once it's available again - at least thats true for a stomp connection
+      ManageIQ::Messaging::Client.open(connect_opts)
+    end
+  end
+
+  def self.add_queue(_meth, ems_id, event)
+    # FIXME: I dont like, we query a deep settings path on every call
+    #  but we want to be able to switch implementations, when config changes via the UI
+    if Settings.workers.worker_base.queue_worker_base.event_handler.queue_type == 'artemis'
+      artemis_client.publish_topic(
+        :service => "events",
+        :sender  => ems_id,
+        :event   => event[:event_type],
+        :payload => event
+      )
+    else
+      MiqQueue.submit_job(
```

Comment:
I had suggested to @agrare that maybe we should just have an MiqQueue backend for manageiq-messaging, which you could then set on first connect.  Then you wouldn't need to check the setting in a loop as you're concerned about.